### PR TITLE
test/docs(sdk): update README and add unit tests

### PR DESCRIPTION
**Pull request by hahattan**
add unit tests for ProvisionWatcher filter logic as a baseline for
each scenario and update the README about the ProvisionWatcher.

Signed-off-by: Chris Hung <chris@iotechsys.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-CommitMessages
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] Other... Please describe:

### CODE REVIEW

File: example/README.md [hunk 1]
```diff
@@ -26,7 +26,8 @@ A Provision Watcher contains the following fields:
 `Profile`: The name of a DeviceProfile which should be assigned to new devices which meet the given criteria  
 `AdminState`: The initial Administrative State for new devices which meet the given criteria  
  
-A candidate new device passes a ProvisionWatcher if all of the Identifiers match, and none of the BlockingIdentifiers.
+A candidate new device passes a ProvisionWatcher if all of the Intifiers match, and none of the BlockingIdentifiers.
```
Comment (lenny-intel) [inline example/README.md:29]: change `Intifiers` to `Identifiers` and `BlockingIdentifiers` to `Blocking Identifiers match`

**Review by lenny-intel — CHANGES_REQUESTED** [review]

**Review by lenny-intel — DISMISSED** [review]
LGTM, @cloudxxx8 ?

**Comment by hahattan** [comment]
rebased

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**